src/Orleans/Streams/Core/IAsyncObservable.cs

```diff
@@ -25,6 +26,7 @@ public interface IAsyncObservable<T>
         /// </returns>
         Task<StreamSubscriptionHandle<T>> SubscribeAsync(IAsyncObserver<T> observer);
 
+        Task<StreamSubscriptionHandle<T>> AttachObserver(IAsyncObserver<T> observer, GuidId subscriptionId);
```

Comment:
The current pattern for attaching observer for existing subscription is to use StreamSubscriptionHandle.Resume.  I don't think we should change that.